crates/commitlog/src/varchar.rs

```diff
@@ -0,0 +1,166 @@
+use std::ops::Deref;
+
+/// An owned string with a max length of `N`, like the venerable VARCHAR.
```

Comment:
I assume this is to implement the `reducer_name` and `outputs` fields as per the proposal? I should have asked this at proposal time, but why not just a string of unbounded length?

Reply:
The `Varchar` type exists because serialization must be infallible -- if that was not the case, a database could be rendered unusable as there is no way to recover from invalid input in a transaction. 

Logging inputs will create huge amounts of repeated data, so there needs to be _some_ limit.

Forcing the caller to prepare data within constraints helps to avoid runtime surprises, and even allows to reject modules early. The slight inconvenience also serves as a nag to come up with a fixed-size stable identifier for the reducer name down the line.